server/resp/src/main/java/org/infinispan/server/resp/commands/string/SETRANGE.java

```diff
@@ -0,0 +1,65 @@
+package org.infinispan.server.resp.commands.string;
+
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.concurrent.CompletionStage;
+
+import org.infinispan.server.resp.ByteBufferUtils;
+import org.infinispan.server.resp.Consumers;
+import org.infinispan.server.resp.Resp3Handler;
+import org.infinispan.server.resp.RespCommand;
+import org.infinispan.server.resp.RespRequestHandler;
+import org.infinispan.server.resp.commands.Resp3Command;
+
+import io.netty.buffer.Unpooled;
+import io.netty.channel.ChannelHandlerContext;
+
+/**
+ * SETRANGE Resp Command
+ *
+ * @link https://redis.io/commands/setrange/
+ * @since 15.0
+ */
+public class SETRANGE extends RespCommand implements Resp3Command {
+   public SETRANGE() {
+      super(4, 1, 1, 1);
+   }
+
+   @Override
+   public CompletionStage<RespRequestHandler> perform(Resp3Handler handler,
+         ChannelHandlerContext ctx,
+         List<byte[]> arguments) {
+      if (arguments.size() != this.getArity() - 1) {
+         ByteBufferUtils.stringToByteBuf("-ERR wrong number of arguments for 'getrange' command\r\n",
+               handler.allocator());
+         return handler.myStage();
+      }
+      byte[] keyBytes = arguments.get(0);
+      int offset = Integer.parseInt(new String(arguments.get(1), StandardCharsets.US_ASCII));
+      byte[] patch = arguments.get(2);
+      CompletionStage<Long> objectCompletableFuture = handler.cache().getAsync(keyBytes)
+            .thenCompose(value -> applyRange(handler, keyBytes, value, patch, offset));
```

Comment:
I think this should be a case for `computeAsync`, instead?

```
CompletionStage<byte[]> cs = handler.cache().computeAsync(keyBytes, (ignore, curr) -> setrange(curr, patch, offset));
```

Reply:
Nice one! Thanks!